src/test/java/racingcar/domain/CarsTest.java

```diff
@@ -0,0 +1,177 @@
+package racingcar.domain;
+
+import static camp.nextstep.edu.missionutils.test.Assertions.*;
+import static org.assertj.core.api.Assertions.*;
+
+import java.io.ByteArrayInputStream;
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import racingcar.view.InputView;
+
+class CarsTest {
+    private static final int MOVING_FORWARD = 4;
+    private static final int STOP = 3;
+
+    private void command(final String... args) {
+        final byte[] buf = String.join("\n", args).getBytes();
+        System.setIn(new ByteArrayInputStream(buf));
+    }
+
+    @Test
+    @DisplayName("Cars가 가지는 값이 불변인지 테스트")
+    void isCarsImmutable() {
+        // given
+        InputView inputView = new InputView();
+        command("pobi,min");
+
+        // when
+        List<String> carNames = inputView.inputCarNames();
+        Cars cars = Cars.from(carNames);
+        carNames.add("aa");
+
+        // then
+        //assertThat(cars.size()).isEqualTo(2);
```

Comment:
두 번째 질문에 대한 답변입니다.

프로덕션 코드에 테스트를 위한 코드를 짜는 것은 좋지 않습니다. 😂
그래도 add, remove, size 같이 개발자가 예상 가능한 메서드는 구현해도 상관없다고 생각합니다.
다른 개발자가 노력님이 만든 클래스를 사용하는 상황을 생각해 봅시다.
해당 클래스를 사용하다가 크기비교가 필요한 경우에`cars.size()` 라는 "메서드가 있을까?" 라고 생각하고 확인해 보고, 해당 메서드가 있으면 "사이즈를 바로 알아낼 수 있겠네?" 라고 생각하고 편하게 사용할 수 있겠죠? 컬렉션에 값을 추가해야할 때, `add()` 라는 메서드가 있는지 확인해보신 경험이 있으실거에요. 이런 상황 때문에 메서드 이름을 컨벤션에 맞춰 직관적으로 네이밍할 필요가 있습니다.
만약 Car 클래스에 `getPositionForTest()` 라는 메서드가 public 으로 존재한다면 사용자는 이 메서드는 무슨 메서드인지 혼동이 올 것입니다.
무슨 메서드인지 어디에 사용하는 메서드인지 모르기 때문에 클래스 파일에 들어와서 메서드가 어떻게 구현되었는 지 로직을 확인해야 하는 상황이 발생해요. 따라서 클래스를 설계할 때는 `기능 동작을 위한 메서드` 혹은 `개발자가 예측가능한 네이밍과 기능의 메서드`만 구현해두시는게 좋습니다.

그리고 현재 구현되어있는 기능으로만 size 테스트를 구현할 경우는 아래와 같이 구현할 수 있겠네요. 👍

```suggestion
        assertThat(cars.findWinnerCarNames()).hasSize(2);
```

사용하지 않는 주석은 제거해주세요! 😉

Reply:
처음 구현할 때 필요없는 메서드라도 `개발자가 예측가능한 네이밍과 기능의 메서드`라면 구현해도 되는 군요! 
테스트코드를 작성하면서 필요한 메서드를 구현하고 다시 지우고 할 필요가 없겠네요!! 
큰 가르침을 얻은 것 같습니다. 
감사드립니다!!